examples/text-classification/run_pl_glue.py

```diff
@@ -1,115 +1,213 @@
+        else:
+            logger.info("No cache file found. Creating one at", cached_dataset_file)
+
+            # Make the directory to hold cached dataset if it doesn't already exist
+            os.makedirs(os.path.dirname(cached_dataset_file), exist_ok=True)
+
+            # We don't know names of text field(s) so we find that here. If multiple, we tokenize text pairs.
+            text_fields = [field.name for field in self.dataset["train"].schema if pyarrow.types.is_string(field.type)]
+
+            def convert_to_features(example_batch):
+                """Function to be mapped across GLUE datasets. Tokenizes text(s) and renames label column.
+
+                Args:
+                    example_batch (Dict[List]): Dict with feature names as keys and batch values as values
+
+                Returns:
+                    Dict[List]: The desired updates to the batch (keys are feature names, values are batch values)
+                """
+
+                # Either encode single sentence or sentence pairs
+                if len(text_fields) > 1:
+                    texts_or_text_pairs = list(zip(example_batch[text_fields[0]], example_batch[text_fields[1]]))
+                else:
+                    texts_or_text_pairs = example_batch[text_fields[0]]
+
+                # Tokenize the text/text pairs
+                features = self.tokenizer.batch_encode_plus(
+                    texts_or_text_pairs, max_length=self.hparams.max_seq_length, pad_to_max_length=True
                 )
-                logger.info("Saving features into cached file %s", cached_features_file)
-                torch.save(features, cached_features_file)
 
-    def load_dataset(self, mode, batch_size):
-        "Load datasets. Called after prepare data."
+                # Rename 'label' to 'labels' so we can directly unpack batches into model call function
+                features["labels"] = example_batch["label"]
 
-        # We test on dev set to compare to benchmarks without having to submit to GLUE server
-        mode = "dev" if mode == "test" else mode
+                return features
 
-        cached_features_file = self._feature_file(mode)
-        logger.info("Loading features from cached file %s", cached_features_file)
-        features = torch.load(cached_features_file)
-        all_input_ids = torch.tensor([f.input_ids for f in features], dtype=torch.long)
-        all_attention_mask = torch.tensor([f.attention_mask for f in features], dtype=torch.long)
-        all_token_type_ids = torch.tensor([f.token_type_ids for f in features], dtype=torch.long)
-        if self.hparams.glue_output_mode == "classification":
-            all_labels = torch.tensor([f.label for f in features], dtype=torch.long)
-        elif self.hparams.glue_output_mode == "regression":
-            all_labels = torch.tensor([f.label for f in features], dtype=torch.float)
+            # TODO - check if this is still right or if it needs updating
```

Comment:
you should not need this anymore because `tokenizer.batch_encode_plus` would return just the relevant attributes according to the tokenizer's (model's) type.

Reply:
I'm using that `cols_to_keep` variable to filter the dataset down to just the kwargs that would be passed to a model's call function. In this case, do you think I still need it? Is there a more elegant way to write it instead?

Reply:
I haven't run your code so I'm not sure. I'm just saying that `tokenizer.batch_encode_plus` should now not return  `token_type_ids` if it's not used in the model afterwards.

Reply:
We can probably just use the map function's remove columns parameter and get rid of the `cols_to_keep` stuff...I'll take a look